src/main/scala/services/mssql/MsSqlConnection.scala

```diff
@@ -170,14 +183,10 @@ class MsSqlConnection(val connectionOptions: ConnectionOptions) extends AutoClos
 
   private type ResultFactory[QueryResultType] = (Statement, ResultSet) => QueryResultType
 
-  private def executeQuery[QueryResultType](query: MsSqlQuery, connection: Connection, resultFactory: ResultFactory[QueryResultType]): Future[QueryResultType] =
-    Future {
-      val statement = connection.createStatement()
-      val resultSet = blocking {
-        statement.executeQuery(query)
-      }
-      resultFactory(statement, resultSet)
-    }
+  private def executeQuery[QueryResultType](query: MsSqlQuery, connection: Connection, resultFactory: ResultFactory[QueryResultType]): Task[QueryResultType] =
+    for statement <- ZIO.attempt(connection.createStatement())
```

Comment:
looks like we are missing scoped close here as well

Reply:
No, it's ok, I added a comment